drivers/regulator/regulator_stm32_vrefbuf.c

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright 2025 CodeWrights GmbH
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT st_stm32_vrefbuf
+
+#include <zephyr/drivers/regulator.h>
+#include <zephyr/drivers/reset.h>
+#include <zephyr/drivers/clock_control/stm32_clock_control.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(regulator_stm32_vrefbuf, CONFIG_REGULATOR_LOG_LEVEL);
+
+struct regulator_stm32_vrefbuf_voltage {
+	uint8_t vrs;
+	uint32_t uv;
+};
```

Comment:
Step 1: change structure
```suggestion
struct regulator_stm32_vrefbuf_voltage {
	uint32_t vrs; /* LL_VREFBUF_VOLTAGE_SCALE_<X> */
	int32_t uv;   /* VREFBUF output in µV with this scale */
};
```